diff -ruN libofx-0.10.2-orig/lib/ofx_preproc.cpp libofx-0.10.2/lib/ofx_preproc.cpp
--- libofx-0.10.2-orig/lib/ofx_preproc.cpp      2021-01-25 22:05:16.000000000 +0100
+++ libofx-0.10.2/lib/ofx_preproc.cpp   2021-10-03 07:22:09.053577003 +0200
@@ -310,7 +310,11 @@
             const char* inchar = s_buffer.c_str();
             char * outchar = iconv_buffer;
             int iconv_retval = iconv (conversion_descriptor,
+#ifdef __MSVC
+                                      &inchar, &inbytesleft,
+#else
                                       const_cast<char**>(&inchar), &inbytesleft,
+#endif
                                       &outchar, &outbytesleft);
             if (iconv_retval == -1)
             {
