SPDX-License-Identifier: BSD-3-Clause

diff --git a/src/winscard_clnt.c b/src/winscard_clnt.c
index b9c900e..4b84ee8 100644
--- a/src/winscard_clnt.c
+++ b/src/winscard_clnt.c
@@ -600,7 +600,22 @@ static LONG SCardEstablishContextTH(DWORD dwScope,
 			goto cleanup;
 		}
 
-		Log3(PCSC_LOG_INFO, "Server is protocol version %d:%d",
+		if (veStr.rv != SCARD_S_SUCCESS){
+			if (veStr.minor == 4 && PROTOCOL_VERSION_MINOR == 5){
+				Log1(PCSC_LOG_INFO, "Reading from server again, mismatch");
+
+				veStr.major = PROTOCOL_VERSION_MAJOR;
+				veStr.minor = 4;
+				veStr.rv = SCARD_S_SUCCESS;
+				rv = MessageSendWithHeader(CMD_VERSION, dwClientID, sizeof(veStr),
+					&veStr);
+
+				Log1(PCSC_LOG_INFO, "Sent to server");
+				rv = MessageReceive(&veStr, sizeof(veStr), dwClientID);
+			}
+		}
+
+		Log3(PCSC_LOG_INFO, "Negotiated protocol version %d:%d",
 			veStr.major, veStr.minor);
 
 		if (veStr.rv != SCARD_S_SUCCESS)
